numpy/_core/src/umath/log1p_complex.h

```diff
@@ -0,0 +1,260 @@
+#ifndef LOG1P_COMPLEX_H
```

Comment:
This is C++ only, I think.  So let's rename it to `.hpp`?

Also please adjust the include guard to be the google-style full file path (see other files).